### PR TITLE
Etcm 139 incomplete data errors during fast sync

### PR DESCRIPTION
**Pull request by kapke**
# Description

Report custom eth error "100 missing data" when getting data from account

# Proposed Solution

implementation of https://eth.wiki/json-rpc/json-rpc-error-codes-improvement-proposal (at least in account-touching methods)

# Important Changes Introduced

Reorganized a bit json serialization. Due to nasty error I run into at some point (json4s and its formats <3) I wanted to plug Encoder-based serialization into rpc servers, due to multiple issues I stopped that and got into working state

# Testing

Call methods like `eth_getBalance` during fast-sync - it should return error with additional information that state node is missing

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 1]
```diff
@@ -867,20 +858,25 @@ class EthService(
...
 
-  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): Either[JsonRpcError, T] = {
-    resolveBlock(blockParam).map { case ResolvedBlock(block, _) =>
-      f(
-        blockchain.getAccount(address, block.header.number).getOrElse(Account.empty(blockchainConfig.accountStartNonce))
-      )
+  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): ServiceResponse[T] =
+    Future {
+      resolveBlock(blockParam).map { case ResolvedBlock(block, _) =>
+        f(
```
Comment (mmrozek) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala:864]: Minor: I would like to name this function `toResponse` or sth similar and reformat the code to
```
val account = blockchain
            .getAccount(address, block.header.number)
            .getOrElse(Account.empty(blockchainConfig.accountStartNonce))
toResponse(account)
```
it is a little bit more readable

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 2]
```diff
@@ -867,20 +858,25 @@ class EthService(
...
+          blockchain
+            .getAccount(address, block.header.number)
+            .getOrElse(Account.empty(blockchainConfig.accountStartNonce))
+        )
+      }
+    }.recover { case _: MissingNodeException =>
```
[the remark this answers is not in the document]
Comment (kapke) [reply src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala:870]: Mainly during fast-sync.

[the remark this answers is not in the document]
Comment (kapke) [reply src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala:870]: It _could_ happen during regular one if state node is lost along the way, but it's much less probable while during fast-sync it is sure that such error happens.

**Review by lemastero — APPROVED** [review]
LGTM 👍

**Pull request merged**